### PR TITLE
Warning fix

### PR DESCRIPTION
**Pull request by mmaret**
Correct few compilation warning

### CODE REVIEW

File: SourceX/dvlnet/tcp_client.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ int tcp_client::create(std::string addrstr, std::string passwd)
 		auto port = default_port;
 		local_server = std::make_unique<tcp_server>(ioc, addrstr, port, passwd);
 		return join(local_server->localhost_self(), passwd);
-	} catch(std::system_error) {
+	} catch(std::system_error &) {
```
Comment (AJenbo) [inline SourceX/dvlnet/tcp_client.cpp]: Lookes like you missed an e
```suggestion
	} catch(std::system_error &e) {
```

File: SourceX/miniwin/misc_msg.cpp [hunk 2]
```diff
@@ -187,7 +187,7 @@ static WPARAM keystate_for_mouse(WPARAM ret)
 
 static WINBOOL false_avail()
 {
-	DUMMY_PRINT("return false although event available", 1);
+	DUMMY_PRINT("return false although event available");
```
Comment (AJenbo) [inline SourceX/miniwin/misc_msg.cpp]: This causes a build error for MSVC

File: SourceX/dvlnet/tcp_client.cpp (hunk 1 above)
Comment (mmaret) [reply SourceX/dvlnet/tcp_client.cpp]: It's not needed but it's too tempting to try github "Commit suggestion" :)

**Comment by mmaret** [comment]
That's a good question. I think that the author was not aware of copy elision optimization done by compiler (with c++98 or above) and wanted to avoid a copy. But as std:move need c++11,  copy elision will be available.

File: SourceX/miniwin/misc_msg.cpp (hunk 2 above)
Comment (mmaret) [reply SourceX/miniwin/misc_msg.cpp]: Ok, gcc got an extension for empty variadic macro (https://gcc.gnu.org/onlinedocs/gcc/Variadic-Macros.html - "empty, the ‘##’ operator causes the preprocessor to remove the comma before it." ) that MSVC do not have. So I'll remove the commit

**Comment by AJenbo** [comment]
Merged manually via https://github.com/diasurgical/devilutionX/commit/19f5686770462644b585ab528b1ad4e69989d177 and https://github.com/diasurgical/devilutionX/commit/89ec0538f51dacc5f94d7d72250f07a2af8ba673

**Pull request closed without merging**